### PR TITLE
feat(Guild): add get_or_fetch_members

### PR DESCRIPTION
**Pull request by EQUENOS**
## Summary

This method is a bulk get-or-fetch method. A detailed description can be found in the docstring.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/guild.py [hunk 1]
```diff
@@ -3311,6 +3312,89 @@ async def query_members(
...
+        member_ids: List[int],
+        *,
+        presences: bool = False,
+        cache: bool = True,
+    ) -> List[Member]:
+        """|coro|
+
+        Tries to get the guild members matching the provided IDs from cache.
+        If some of them were not found, the method requests the missing members using websocket operations.
+        If ``cache`` kwarg is ``True`` (default value) the missing members are cached.
```
Comment (Victorsitou) [inline disnake/guild.py:3326]:
```suggestion
        If ``cache`` kwarg is ``True`` (default value) the missing members will be cached.
```

**Review by Victorsitou — COMMENTED** [review]
Other than that, looks good to me

File: disnake/guild.py [hunk 2]
```diff
@@ -3311,6 +3312,89 @@ async def query_members(
             self, query=query, limit=limit, user_ids=user_ids, presences=presences, cache=cache
         )
 
+    async def get_or_fetch_members(
+        self,
+        member_ids: List[int],
```
Comment (shiftinv) [inline disnake/guild.py:3317]: Maybe change this to `user_ids` to keep it consistent with `Guild.query_members` and the api docs.

File: disnake/guild.py [hunk 3]
```diff
@@ -3311,6 +3312,89 @@ async def query_members(
...
+        Websocket operations can be slow, however, this method is cheaper than multiple :meth:`get_or_fetch_member` calls.
+
+        .. versionadded:: 2.4
+
+        Parameters
+        -----------
+        member_ids: List[:class:`int`]
+            List of user IDs to search for. If the user ID is not in the guild then it won't be returned.
+        presences: :class:`bool`
+            Whether to request for presences to be provided. This defaults to ``False``.
```
Comment (shiftinv) [inline disnake/guild.py:3339]:
```suggestion
            Whether to request for presences to be provided. Defaults to ``False``.
```

File: disnake/guild.py [hunk 4]
```diff
@@ -3311,6 +3312,89 @@ async def query_members(
...
+        presences: bool = False,
+        cache: bool = True,
+    ) -> List[Member]:
+        """|coro|
+
+        Tries to get the guild members matching the provided IDs from cache.
+        If some of them were not found, the method requests the missing members using websocket operations.
+        If ``cache`` kwarg is ``True`` (default value) the missing members will be cached.
+
+        If more than 100 members are missing, several websocket operations are made. Otherwise only one.
```
Comment (shiftinv) [inline disnake/guild.py:3328]:
```suggestion
        If more than 100 members are missing, several websocket operations are made.
```

File: disnake/guild.py [hunk 5]
```diff
@@ -3311,6 +3312,89 @@ async def query_members(
...
+        -------
+        asyncio.TimeoutError
+            The query timed out waiting for the members.
+        ClientException
+            The presences intent is not enabled.
+
+        Returns
+        --------
+        List[:class:`Member`]
+            The list of members that have matched the IDs.
```
Comment (shiftinv) [inline disnake/guild.py:3355]:
```suggestion
            The list of members with the given IDs, if they exist.
```

File: disnake/guild.py [hunk 6]
```diff
@@ -3311,6 +3312,89 @@ async def query_members(
...
+        if len(unresolved_ids) == 1:
+            # fetch_member is cheaper than query_members
+            try:
+                members.append(await self.fetch_member(unresolved_ids[0]))
+            except HTTPException:
+                pass
+        else:
+            # We have to split the request into several smaller requests
+            # because the limit is 100 members per request.
+            # Imo no one should ever request more than 100 members using this method.
```
Comment (shiftinv) [inline disnake/guild.py:3382]: why not?

**Review by shiftinv — COMMENTED** [review]
nice! just a few nitpicks, primarily documentation-related

File: disnake/guild.py (hunk 6 above)
Comment (EQUENOS) [reply disnake/guild.py:3382]: I can't think of a use case where you need to fetch more than 100 members. People usually fetch members to get full tags in order to display them in a nice leaderboard or something like that, so 10-20 users should be enough. But that's just my opinion based on intuition. I will remove this line, it's confusing.

**Review by shiftinv — APPROVED** [review]

**Pull request merged**